### PR TITLE
Chain simulator tests refactor

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- refactor for chain simulator tests to be more general (and to work better with sovereign chain simulator)
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by mariusmihaic — APPROVED** [review]

File: integrationTests/chainSimulator/interface.go [hunk 1]
```diff
@@ -3,11 +3,12 @@ package chainSimulator
 import (
 	"math/big"
 
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/process"
+
```
Comment (miiu96) [inline integrationTests/chainSimulator/interface.go:8]: remove empty line

File: integrationTests/chainSimulator/staking/common.go [hunk 2]
```diff
@@ -5,24 +5,17 @@ import (
...
 
-	"github.com/multiversx/mx-chain-core-go/core"
-	"github.com/multiversx/mx-chain-core-go/data/transaction"
 	chainSimulatorIntegrationTests "github.com/multiversx/mx-chain-go/integrationTests/chainSimulator"
-	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
 	chainSimulatorProcess "github.com/multiversx/mx-chain-go/node/chainSimulator/process"
 	"github.com/multiversx/mx-chain-go/process"
 	"github.com/multiversx/mx-chain-go/vm"
+
```
Comment (miiu96) [inline integrationTests/chainSimulator/staking/common.go:13]: remove empty line

File: integrationTests/chainSimulator/staking/stake/simpleStake_test.go [hunk 3]
```diff
@@ -7,18 +7,19 @@ import (
...
 	"github.com/multiversx/mx-chain-go/common"
 	"github.com/multiversx/mx-chain-go/config"
+	chainSimulatorIntegrationTests "github.com/multiversx/mx-chain-go/integrationTests/chainSimulator"
 	"github.com/multiversx/mx-chain-go/integrationTests/chainSimulator/staking"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
 	"github.com/multiversx/mx-chain-go/node/chainSimulator/process"
 	"github.com/multiversx/mx-chain-go/vm"
+
```
Comment (miiu96) [inline integrationTests/chainSimulator/staking/stake/simpleStake_test.go:19]: remove empty line

File: integrationTests/chainSimulator/staking/stake/stakeAndUnStake_test.go [hunk 4]
```diff
@@ -26,6 +19,13 @@ import (
 	chainSimulatorProcess "github.com/multiversx/mx-chain-go/node/chainSimulator/process"
 	"github.com/multiversx/mx-chain-go/process"
 	"github.com/multiversx/mx-chain-go/vm"
+
```
Comment (miiu96) [inline integrationTests/chainSimulator/staking/stake/stakeAndUnStake_test.go:22]: also here

File: integrationTests/chainSimulator/testing.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+	"encoding/base64"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/errors"
+	chainSimulatorProcess "github.com/multiversx/mx-chain-go/node/chainSimulator/process"
+	"github.com/multiversx/mx-chain-go/process"
+
```
Comment (miiu96) [inline integrationTests/chainSimulator/testing.go:13]: remove empty line

**Review by miiu96 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**